src/utils/server-utils.ts

```diff
@@ -0,0 +1,165 @@
+import moment from 'moment';
+import { xml2js } from 'xml-js';
+import { Map } from 'immutable';
+import { merge, unset, get, isString, union, isEmpty } from 'lodash';
+import { format, parse } from 'url';
+
+import config from '../config/prism.json';
+import { AvailableDates } from '../config/types';
+
+const xml2jsOptions = {
+  compact: true,
+  trim: true,
+  ignoreComment: true,
+};
+
+export function formatServerUri(
+  serverUri: string,
+  queryProp: { [key: string]: string | boolean | number },
+  selectedDate?: number,
+) {
+  // The second arg of 'parse' allows us to have 'query' as an object
+  const { query, ...parsedUrl } = parse(serverUri, true);
+
+  // Removing 'search' to be able to format by 'query'
+  unset(parsedUrl, 'search');
+
+  const selectedDateString = selectedDate
+    ? moment(selectedDate).format('YYYY-MM-DD')
+    : '';
+
+  return decodeURI(
+    format({
+      ...parsedUrl,
+      query: merge(query, queryProp, { time: selectedDateString }),
+    }),
+  );
+}
+
+/**
+ * Format the raw data to { [layerId]: availableDates }
+ * @param rawLayers Layers data return by the server 'GetCapabilities' request
+ * @param layerIdPath path to layer's id
+ * @param datesPath path to layer's available dates
+ * @returns an object shape like { [layerId]: availableDates }
+ */
+function formatCapabilitiesInfo(
+  rawLayers: any,
```

Comment:
You don't really need to explicitly define `any` types - that's the default. In general, you should _try_ to do stronger typing than `any`, but if it's not possible (or worth it), you don't need to actually type them as `any`

Reply:
Yeah, I think it's not worth giving a type to `rawLayers` because it comes from the server and it is formatted as soon as we received it.

Should I turn off the `strict mode` on `tsconfig.json`? 
The strict mode automatically turn on `noImplicitAny`
![image](https://user-images.githubusercontent.com/40799514/81454150-bf0cf300-918b-11ea-87dd-86ee88c66270.png)

I kinda prefer explicit any.